xwayland/sockets.c

```diff
@@ -0,0 +1,147 @@
+#define _XOPEN_SOURCE 700
```

Comment:
We have quite a few #define _XOPEN_SOURCE around, should we put it in some global configuration? (in meson as -D or a global header?)